### PR TITLE
ED19 and ECR1 benchmarks

### PR DESCRIPTION
**Pull request by Dentosal**
Closes https://github.com/FuelLabs/fuel-core/issues/1254

### CODE REVIEW

**Comment by xgreenx** [comment]
Could you also update `block_target_gas.rs` benchmark and add cases for ED19 and ECR1, please?=) An infinite loop that calls each of them until you spend whole gas.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Description of the upcoming release here.
 
 - Something new here 1
 - Something new here 2
+- [#1263](https://github.com/FuelLabs/fuel-core/pull/1263): Add gas benchmarks for `ED19` and `ECR1` instructions
```
Comment (xgreenx) [inline CHANGELOG.md:15]:
```suggestion
- [#1263](https://github.com/FuelLabs/fuel-core/pull/1263): Add gas benchmarks for `ED19` and `ECR1` instructions.
```

**Review by xgreenx — APPROVED** [review]

**Comment by Dentosal** [comment]
> Could you also update `block_target_gas.rs` benchmark and add cases for ED19 and ECR1, please?=) An infinite loop that calls each of them until you spend whole gas.

Yep, added in 2506a03.

File: benches/benches/set/crypto.rs [hunk 2]
```diff
@@ -15,48 +16,135 @@ use rand::{
...
+                op::aloc(0x10),
+                op::move_(0x11, RegId::HP),
+            ])
+            .with_data(
+                ed19_signature
+                    .to_bytes()
+                    .iter()
+                    .chain(message.iter())
+                    .copied()
+                    .collect(),
```
Comment (xgreenx) [inline benches/benches/set/crypto.rs:147]: It seems you forgot public key

File: benches/benches/set/crypto.rs [hunk 3]
```diff
@@ -15,48 +16,135 @@ use rand::{
...
+                op::addi(
+                    0x21,
+                    0x20,
+                    ed19_keypair.public.as_ref().len().try_into().unwrap(),
+                ),
+                op::addi(
+                    0x22,
+                    0x21,
+                    ed19_signature.as_ref().len().try_into().unwrap(),
+                ),
```
Comment (xgreenx) [inline benches/benches/set/crypto.rs:135]: Both `addi` write into `0x22`, so you will override the information

File: benches/benches/set/crypto.rs [hunk 4]
```diff
@@ -15,48 +16,135 @@ use rand::{
...
+                op::addi(0x22, 0x21, message.as_ref().len().try_into().unwrap()),
+                op::movi(0x10, PublicKey::LEN.try_into().unwrap()),
+                op::aloc(0x10),
+                op::move_(0x11, RegId::HP),
```
Comment (xgreenx) [inline benches/benches/set/crypto.rs:139]: Do we need this part?=)

File: benches/benches/set/crypto.rs [hunk 5]
```diff
@@ -15,48 +16,135 @@ use rand::{
...
+        "ecr1",
+        VmBench::new(op::ecr1(0x11, 0x20, 0x21))
+            .with_prepare_script(vec![
+                op::gtf_args(0x20, 0x00, GTFArgs::ScriptData),
+                op::addi(
+                    0x21,
+                    0x20,
+                    ecr1_signature.as_ref().len().try_into().unwrap(),
+                ),
+                op::addi(0x22, 0x21, message.as_ref().len().try_into().unwrap()),
```
Comment (xgreenx) [inline benches/benches/set/crypto.rs:102]: You don't use register `0x22`

File: benches/benches/set/crypto.rs [hunk 6]
```diff
@@ -15,48 +16,135 @@ use rand::{
...
     let rng = &mut StdRng::seed_from_u64(2322u64);
 
-    let secret = SecretKey::random(rng);
     let message = Message::new(b"foo");
-    let signature = Signature::sign(&secret, &message);
+
+    let eck1_secret = SecretKey::random(rng);
+    let eck1_signature = Signature::sign(&eck1_secret, &message);
 
     run_group_ref(
```
Comment (xgreenx) [inline benches/benches/set/crypto.rs:24]: Can we keep `ecr1` near `eck1` please?=)

File: benches/benches/set/crypto.rs [hunk 7]
```diff
@@ -15,48 +16,135 @@ use rand::{
...
         VmBench::new(op::eck1(0x11, 0x20, 0x21))
             .with_prepare_script(vec![
                 op::gtf_args(0x20, 0x00, GTFArgs::ScriptData),
-                op::addi(0x21, 0x20, signature.as_ref().len().try_into().unwrap()),
+                op::addi(
+                    0x21,
+                    0x20,
+                    eck1_signature.as_ref().len().try_into().unwrap(),
+                ),
                 op::addi(0x22, 0x21, message.as_ref().len().try_into().unwrap()),
```
Comment (xgreenx) [inline benches/benches/set/crypto.rs:35]: We don't use `0x22`

File: benches/benches/set/crypto.rs (hunk 4 above)
Comment (Dentosal) [reply benches/benches/set/crypto.rs]: Removed in 07f6d107

File: benches/benches/set/crypto.rs (hunk 5 above)
Comment (Dentosal) [reply benches/benches/set/crypto.rs]: Removed in 07f6d107

File: benches/benches/set/crypto.rs (hunk 7 above)
Comment (Dentosal) [reply benches/benches/set/crypto.rs]: Removed in 07f6d107

File: benches/benches/set/crypto.rs (hunk 2 above)
Comment (Dentosal) [reply benches/benches/set/crypto.rs]: d3336c4c

File: benches/benches/set/crypto.rs [hunk 8]
```diff
@@ -15,48 +16,129 @@ use rand::{
...
+                op::gtf_args(0x20, 0x00, GTFArgs::ScriptData),
+                op::addi(
+                    0x21,
+                    0x20,
+                    ecr1_signature.as_ref().len().try_into().unwrap(),
+                ),
                 op::addi(0x22, 0x21, message.as_ref().len().try_into().unwrap()),
                 op::movi(0x10, PublicKey::LEN.try_into().unwrap()),
                 op::aloc(0x10),
                 op::move_(0x11, RegId::HP),
```
Comment (xgreenx) [inline benches/benches/set/crypto.rs:65]:
```suggestion
        VmBench::new(op::ecr1(RegId::HP, 0x20, 0x21))
            .with_prepare_script(vec![
                op::gtf_args(0x20, 0x00, GTFArgs::ScriptData),
                op::addi(
                    0x21,
                    0x20,
                    ecr1_signature.as_ref().len().try_into().unwrap(),
                ),
                op::movi(0x10, PublicKey::LEN.try_into().unwrap()),
                op::aloc(0x10),
```

**Review by xgreenx — APPROVED** [review]

**Pull request merged**